convert/process_csv.py

```diff
@@ -0,0 +1,88 @@
+import pandas as pd
+from collections import deque
+import sys
+
+
+def addExtension(tVal):
+    return str(tVal) + ".png"
+
+
+if __name__ == '__main__':
+    dataDir = sys.argv[1]
+    timestampPath = sys.argv[2]
+    gyroPath = sys.argv[3]
+
+    # ----------#
+
+    timestampLabels = ["#timestamp [ns]", "filename"]
+    timestamps = pd.read_csv(timestampPath, header=None, dtype=str)
+    outputFilename = dataDir + "/timestamps.csv"
+
+    data = {
+        timestampLabels[0]: timestamps[0], timestampLabels[1]: timestamps[0] + ".png"
+    }
+
+    dataframe = pd.DataFrame(data=data)
+    dataframe.to_csv(outputFilename, sep=',', encoding='utf-8', index=False, line_terminator='\r\n')
+
+    # ----------#
+
+    imuLabels = [
+        "#timestamp [ns]",
+        "w_RS_S_x [rad s^-1]",
+        "w_RS_S_y [rad s^-1]",
+        "w_RS_S_z [rad s^-1]",
+        "a_RS_S_x [m s^-2]",
+        "a_RS_S_y [m s^-2]",
+        "a_RS_S_z [m s^-2]"
+    ]
+
+    gyro = pd.read_csv(gyroPath, header=None)
+    outputFilename = dataDir + "/gyro_accel.csv"
+
+    if len(sys.argv) == 5:
+        accelPath = sys.argv[4]
+        accel = pd.read_csv(accelPath, header=None)
+
+        tempVal = 0
+        index = 0
+        counter = 0
+
+        accelX = deque()
+        accelY = deque()
+        accelZ = deque()
+
+        for indexGyro, rowGyro in gyro.iterrows():
+            for i in range(index, len(accel)):
+                if accel.loc[i, 3] > rowGyro[0]:
+                    index = i + 1
```

Comment:
Also this ```index``` value seems useless because range() is like a list created once at the start

If this is intended to solve mismatch between accel/gyro timestamps we could try standard pandas function merge_asof (merge by nearest) on timestamp?

Reply:
Yes, I suppose merge_asof should solve that (I need to read about it)